echopype/consolidate/api.py

```diff
@@ -67,66 +70,183 @@ def add_depth(
     ds : xr.Dataset
         Source Sv dataset to which a depth variable will be added.
         Must contain `echo_range`.
-    depth_offset : float
+    echodata : EchoData, default=None
+        ``EchoData`` object from which the ``Sv` dataset originated.
+        It must contain transducer position and orientation information.
+    echodata_strict : bool, default=False
+        If set to True and EchoData object is passed, check for invalid
+        or suspect values in the source data arrays.
+        CURRENTLY NOT BEING USED.
+    depth_offset : float, default=0
         Offset along the vertical (depth) dimension to account for actual transducer
         position in water, since `echo_range` is counted from transducer surface.
-        Default is 0.
-    tilt : float
-        Transducer tilt angle [degree].
-        Default is 0 (transducer mounted vertically).
-    downward : bool
-        Whether or not the transducers point downward.
-        Default to True.
+    tilt : float, default=0
+        Transducer tilt angle [degree]. 0 corresponds to a transducer pointing vertically.
+    downward : bool, default=True
+        The transducers point downward.
 
     Returns
     -------
     The input dataset with a `depth` variable (in meters) added
 
     Notes
     -----
+    See https://echopype.readthedocs.io/en/stable/data-proc-additional.html#vertical-coordinate-z-axis-variables # noqa
+
     Currently this function only scalar inputs of depth_offset and tilt angle.
     In future expansion we plan to add the following options:
 
     * Allow inputs as xr.DataArray for time-varying variations of these variables
     * Use data stored in the EchoData object or raw-converted file from which the Sv is derived,
       specifically `water_level`, `vertical_offtset` and `tilt` in the `Platform` group.
     """
-    # TODO: add options to use water_depth, vertical_offset, tilt stored in EchoData
-    # # Water level has to come from somewhere
-    # if depth_offset is None:
-    #     if "water_level" in ds:
-    #         depth_offset = ds["water_level"]
-    #     else:
-    #         raise ValueError(
-    #             "water_level not found in dataset and needs to be supplied by the user"
-    #         )
-
-    # # If not vertical needs to have tilt
-    # if not vertical:
-    #     if tilt is None:
-    #         if "tilt" in ds:
-    #             tilt = ds["tilt"]
-    #         else:
-    #             raise ValueError(
-    #                 "tilt not found in dataset and needs to be supplied by the user. "
-    #                 "Required when vertical=False"
-    #             )
-    # else:
-    #     tilt = 0
 
```

Comment:
```suggestion
    del echodata_strict

```
Looks like pre-commit is currently not throwing error for unused argument! Interesting... anyhow, maybe it's better to just remove it here for now as a better practice: https://pycodequ.al/docs/pylint-messages/w0613-unused-argument.html#how-to-fix

Reply:
Right now that argument is more for discussion; ie, whether we need something like it. If we end up seeing value in it, it will be implemented (used). Otherwise, it'll be dropped. So, I won't change anything yet.